### ISSUE TITLE
Paper Feedback: Analytic methods for derivative calculations

### ISSUE THREAD

**Issue by hayesall** [elwinter/nnde#16]
The "Statement of need" (lines 19-22) mentioned that existing deep learning frameworks can speed up computation, but **cannot** perform analytical derivatives needed for the trial function method implemented by `nnde`.

![Screenshot of the paper, where it shows something similar to the text summarized above.](https://user-images.githubusercontent.com/11916674/125535312-75146622-1797-40e0-b50a-0b63a3a49f52.png)

Can this point be clarified or expanded?

For example: PyTorch implements methods for automatic differentiation ([autograd](https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html)) and numerical methods to compute gradients. What is an example where these are insufficient?

**Comment by byekrang** [comment]
Would you please explain about extra functionalities of nnde with this version of [Autograd](https://github.com/HIPS/autograd) (not the PyTorch's one) and [JAX](https://github.com/google/jax) considering [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)?

**Comment by elwinter** [comment]
We updated the paper to address this issue by describing how the code was started before the widespread adoption of TensorFlow 2. We began to convert the software to use TensorFlow 2 several months ago, and early results are promising, so we wanted to document this particular stage in the project prior to further development.
